pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -185,7 +196,17 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 				if !ok {
 					logger.Infof("orchestration status config map result channel closed, will restart watch.")
 					w.Stop()
-					<-time.After(100 * time.Millisecond)
+					<-time.After(5 * time.Second)
+					leftNodes := 0
+					leftNodes, remainingNodes, completed, statuses, err = c.checkNodesCompleted(selector, config, configOSDs)
+					if err == nil {
+						if completed {
+							logger.Infof("additional %d/%d node(s) completed osd provisioning", leftNodes, originalNodes)
+							return true
+						}
+					} else {
+						logger.Warningf("failed to list orchestration configmap, status: %v", err)
```

Comment:
i'm not sure if just a warning is the right thing to do here, because the call on line 201 to `checkNodesCompleted` would have wiped out all our state tracking in `remainingNodes` and `statuses`.  If we just log as a warning here then break the `ResultLoop` and start at the watch again, we have no state to keep making decisions on, we'll think there are 0 remaining nodes and that is not true.  Am I reading this right? what do you think we can do to harden against that?